website/docs/r/schema_site_contract_service_graph_listener.html.markdown

```diff
@@ -0,0 +1,178 @@
+  listener_name      = "example"
+  port               = 80
+  protocol           = "udp"
+  schema_id          = mso_schema.example.id
+  security_policy    = "default"
+  service_node_index = 0
+  site_id            = mso_schema_site_service_graph.example.site_id
+  template_name      = one(mso_schema.example.template).name
+  rules {
+    action_type       = "forward"
+    content_type      = "textPlain"
+    name              = "default"
+    port              = 80
+    priority          = 0
+    protocol          = "udp"
+    redirect_code     = "permMoved"
+    redirect_port     = 0
+    redirect_protocol = "inherit"
+    target_ip_type    = "unspecified"
+    url_type          = "original"
+    health_check {
+      interval            = 5
+      port                = 80
+      protocol            = "tcp"
+      success_code        = "200-399"
+      timeout             = 0
+      unhealthy_threshold = 2
+      use_host_from_rule  = "no"
+    }
+    provider_epg_ref {
+      anp_name      = mso_schema_template_anp.example.name
+      epg_name      = mso_schema_template_anp_epg_contract.example.epg_name
+      schema_id     = mso_schema.example.id
+      template_name = one(mso_schema.example.template).name
+    }
+  }
+}
+
+```
+
+## Argument Reference ##
+
+* `schema_id` - (Required) The schema ID of the Listener.
+* `template_name` - (Required) The template name of the Listener.
+* `contract_name` - (Required) The contract name of the Listener.
+* `site_id` - (Required) The site ID under which the Listener is deployed.
+* `service_node_index` - (Required) The service node index of the Service Graph under which the Listener is deployed.
+* `listener_name` - (Required) The name of the Listener.
+* `protocol` - (Required) The protocol of the Listener. Allowed values are `http`, `https`, `tcp`, `udp`, `tls` and `inherit`.
+* `port` - (Required) The port number of the Listener.
+* `security_policy` - (Optional) The security policy of the Listener. Allowed values are `default`, `eLBSecurityPolicy-2016-08`, `eLBSecurityPolicy-FS-2018-06`, `eLBSecurityPolicy-TLS-1-2-2017-01`, `eLBSecurityPolicy-TLS-1-2-Ext-2018-06`, `eLBSecurityPolicy-TLS-1-1-2017-01`, `eLBSecurityPolicy-2015-05`, `eLBSecurityPolicy-TLS-1-0-2015-04`, `AppGwSslPolicyDefault`, `AppGwSslPolicy20150501`, `AppGwSslPolicy20170401` and `AppGwSslPolicy20170401S`.
+* `frontend_ip_dn` - (Optional) The frontend IP DN of the Cloud L4-L7 Network Load Balancer device. The frontend IP should be configured in the Azure Cloud Network Controller. The frontend IP configuration option will be available under the `Advanced Settings` of the Network Load Balancer device.
+* `ssl_certificates` - (Optional) The SSL Certificates information of the Listener.
+  * `name` - (Required) The key ring name of the SSL Certificate.
+  * `target_dn` - (Required) The key ring DN of the SSL Certificate.
+  * `default` - (Read-Only) The default boolean flag of the SSL Certificate.
```

Comment:
Should this be read-only?

Reply:
As per the offline discussion, removed from the schema.